pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -161,8 +142,11 @@ def advance(self, dataloader_iter: Iterator, **kwargs: Any) -> None:
         processed_batch_end_outputs = self._prepare_outputs(batch_end_outputs, batch_mode=True)
 
         # hook
-        if not isinstance(self.batch_loop, IteratorBatchProcessor):
-            self.trainer.call_hook("on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, 0)
+        self.trainer.call_hook("on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, 0)
+
+        # free memory
+        del batch
+
```

Comment:
n00b question: in the case of `DataLoaderIterDataFetcher`  - is the `batch` the data iterator? if so, wouldn't deleting this create issues?

Reply:
Good catch !